src/ctests/all_native_events.c

```diff
@@ -22,169 +23,199 @@
+    }
+
+    /* Add the event */
+    retval = PAPI_add_event( EventSet, event_code );
+    if ( retval != PAPI_OK ) {
+        if (!quiet) printf( "Error adding %s %d\n", name, retval );
+            return retval;
+    }
+
+    /* Start the event */
+    retval = PAPI_start( EventSet );
+    if ( retval != PAPI_OK ) {
+        PAPI_perror( "PAPI_start" );
+    } else {
+        retval = PAPI_stop( EventSet, &values );
+        if ( retval != PAPI_OK ) {
+            PAPI_perror( "PAPI_stop" );
+            return retval;
+        } else {
+            if (!quiet) printf( "Added and Stopped %s successfully.\n", name );
+        }
+     }
+
+     /* Cleanup the eventset */
+     retval=PAPI_cleanup_eventset( EventSet );
+     if (retval != PAPI_OK ) {
+         test_fail( __FILE__, __LINE__, "PAPI_cleanup_eventset", retval);
+     }
+
+     /* Destroy the eventset */
+     retval=PAPI_destroy_eventset( &EventSet );
+     if (retval != PAPI_OK ) {
+         test_fail( __FILE__, __LINE__, "PAPI_destroy_eventset", retval);
+     }
+
+     return PAPI_OK;
+}
+
+static void
+print_help(char **argv)
+{
+    printf( "This is the all_native_events program.\n" );
+    printf( "For all components compiled in, it attempts to: add the component events, start, and stop.\n" );
+    printf( "Usage: %s [options]\n", argv[0] );
+    printf( "General command options:\n" );
+    printf( "\t-h, --help                           Print the help message.\n" );
+    printf( "\t--disable-cuda-events=<yes,no>       Optionally disable processing the Cuda native events. Default is no.\n" );
+    printf( "\n" );
 }
 
 int
 main( int argc, char **argv )
 {
 
-	int i, k, add_count = 0, err_count = 0;
-	int retval;
-	PAPI_event_info_t info, info1;
-	const PAPI_hw_info_t *hwinfo = NULL;
-	const PAPI_component_info_t* cmpinfo;
-	int event_code;
-	int numcmp, cid;
-	int quiet;
-
-	/* Set quiet variable */
-	quiet=tests_quiet( argc, argv );
-
-	/* Init PAPI library */
-	retval = PAPI_library_init( PAPI_VER_CURRENT );
-	if ( retval != PAPI_VER_CURRENT ) {
-		test_fail( __FILE__, __LINE__, "PAPI_library_init", retval );
-	}
-
-	if (!quiet) {
-		printf("Test case ALL_NATIVE_EVENTS: Available "
-				"native events and hardware "
-				"information.\n");
-	}
-
-	hwinfo=PAPI_get_hardware_info();
-	if ( hwinfo == NULL ) {
-		test_fail( __FILE__, __LINE__, "PAPI_get_hardware_info", 2 );
-	}
-
-	numcmp = PAPI_num_components(  );
+    int i, k, add_count = 0, err_count = 0;
+    int retval;
+    PAPI_event_info_t info, info1;
+    const PAPI_hw_info_t *hwinfo = NULL;
+    const PAPI_component_info_t* cmpinfo;
+    char disableCudaEvts[PAPI_MIN_STR_LEN] = "no";
+    int event_code;
+    int numcmp, cid;
+    int quiet;
+
+    /* Set quiet variable */
+    quiet=tests_quiet( argc, argv );
+
+    /* parse command line flags */
+    for (i = 0; i < argc; i++) {
+        if (strncmp(argv[i], "--disable-cuda-events=", 22) == 0) {
+            strcpy(disableCudaEvts, argv[i] + 22);
```

Comment:
It would be better to use: strcpy(disableCudaEvts, argv[i] + 22, PAPI_MIN_STR_LEN);

Reply:
I have updated `all_native_events.c` to use `strncpy`.